Hide duplicated header and "Add new _" button in product & coupon editor

tomalec opened the pull request:
_:warning: This is an addon/ follow up to https://github.com/woocommerce/woocommerce/pull/52632_

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

As part of the Product editor UI polish, @j111q proposed to remove the duplicated header and "Add new _" button.
I believe such removal could be a bit more disruptive to extensions tweaking around the header or merchants that got used to this flow. So, I made a separate PR to emphasize it more in git history, added a separate changelog, and gave it a more dedicated review and testing.


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout and build this branch
2. Go to 
	1. Edit product page
	2. Edit coupon page 	
4. Check that there is no header and "add new _" button visible and the layout look good and as in jqmi03xIh4d0USEUM6U8jZ-fi-4_166
5. Resize the view port to test responsiveness
6. Open screen options
7. Bring up some notice, or mimic that by `document.querySelector('#lost-connection-notice').style.display = 'block';`


#### Before
![image](https://github.com/user-attachments/assets/fc49fd1d-1c73-4391-a2a5-96f7f025b64d)
#### After
![image](https://github.com/user-attachments/assets/d78abdee-28b0-4ef2-8e68-1d9f85bdfb09)


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -93,6 +116,12 @@ $break-largeish: 900px; // It's almost $break-large, but kept consistent with th
 	}
 
+	#poststuff {
+		// Add collapsable gap between post form and notices.
```
```suggestion
		// Add collapsible gap between post form and notices.
```

eason9487 commented on plugins/woocommerce/client/legacy/css/admin/product-editor.scss:
```diff
@@ -70,9 +70,32 @@ $break-largeish: 900px; // It's almost $break-large, but kept consistent with th
+		 * We could have clear the floats on .wrap container, which would make margin collapsing easier.
+		 * But, we clear it on hr to still allow extensions to inject something next to the header, or displaying it back again.
+		 */
+		 .wp-header-end {
```
```suggestion
		.wp-header-end {
```

eason9487 reviewed (approved):
Thanks for the work! LGTM.

tomalec committed:
Hide duplicated header and Add new button

Reset branch, to update with squashed target.

Commits:

ce46f2ba43158750add04fe981778f971d8cb983
Hide duplicated header and Add new button

1bd9da20592e3cf5d750e6692c2e686c9ef01b90
Add changelog entry

fe2de3a170fd09ff96d0e68820a4d119622d0a99
move rule around to keep them more aligned with the document order

0ed32d024c5002e5ebb7972b74052ea921df545c
Apply suggestions from code review

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec commented:
I force-pushed reset branch, as after **squashed** merging target branch to trunk, this PR turned out to be 100+ commits and full of conflicts.

As GitHub shows above, the effective results are identical.
